lib/fluent/plugin/filter_sampling.rb

```diff
@@ -54,4 +57,15 @@ def filter_with_minimum_rate(t, record)
       nil
     end
   end
+
+  def record_key(tag, record)
+    case @sample_unit
+    when :all
+      'all'
+    when 'tag'
+      tag
+    else
+      @accessor.call(record)
+    end
+  end
```

Comment:
This seems not to work expectedly because this changes the parameter type from enum to string (L9). The value of an enum is symbol, but the value of a string is a string. If the value will be `all`, this method will return the accessor for `all` column. It should be unexpected behavior.

Reply:
Yep, that was a silly mistake. I've fixed it and added a test for the 'all' behavior as well.